status/flowcell.py

```diff
@@ -109,3 +114,255 @@ def get(self, flowcell_id):
+                lambda x: round(x*100, 2) if round(x*100, 2) > 0 else "")
+            df["basecalled_pass_bases_pc"] = (df.basecalled_pass_bases / sum(df.basecalled_pass_bases)).apply(
+                lambda x: round(x*100, 2) if round(x*100, 2) > 0 else "")
+            # Calculate new metrics
+            df["average_read_length_passed"] =  (df.basecalled_pass_bases / df.basecalled_pass_read_count).apply(
+                lambda x: int(round(x, 0)))
+            df["accuracy"] =  (df.basecalled_pass_bases / (df.basecalled_pass_bases + df.basecalled_fail_bases)).apply(
+                lambda x: round(x*100, 2) if round(x*100, 2) > 0 else "")
+
+            # Return dict for easy Tornado templating
+            df.fillna("", inplace=True)
+            bcs = df.to_dict(orient="index")
+            return bcs
+        
+        else:
+            return None
+    
+    def fetch_args(self, name):
+        view_args = self.application.nanopore_runs_db.view("info/args", descending=True)
+        l = [row for row in view_args.rows if name in row.key][0].value
+
+        group = "([^\s=]+)"
+        flag_pair = re.compile(f"^--{group}={group}$")
+        flag_key_or_header = re.compile(f"^--{group}$")
+        val = re.compile(f"^(?!--){group}$")
+        pair = re.compile(f"^(?!--){group}={group}$")
+
+        entries = {}
+        idxs_args = iter(zip(range(0, len(l)), l))
+        for (idx, arg) in idxs_args:
+            # Flag pair --> Tuple
+            if re.match(flag_pair, arg):
+                k, v = re.match(flag_pair, arg).groups()
+                entries[(k,v)]="pair"
+            # Flag single
+            elif re.match(flag_key_or_header, arg):
+                # If followed by pair --> Header
+                if re.match(pair, l[idx+1]):
+                    h = re.match(flag_key_or_header, arg).groups()[0]
+                    entries[h]="header"
+                # If followed by val --> Add both as tuple and skip ahead
+                elif re.match(val, l[idx+1]):
+                    k = re.match(flag_key_or_header, arg).groups()[0]
+                    v = re.match(val, l[idx+1]).groups()[0]
+                    entries[(k,v)]="pair"
+                    next(idxs_args)
+            # Pair 
+            elif re.match(pair, arg):
+                k, v = re.match(pair, arg).groups()
+                entries[(k,v)]="sub_pair"
+
+        return entries
+
+    def get(self, name):
+
+        t = self.application.loader.load("ont_flowcell.html")
+        self.write(t.generate(gs_globals=self.application.gs_globals,
+                                flowcell=self.fetch_ont_flowcell(name),
+                                barcodes=self.fetch_barcodes(name),
+                                args=self.fetch_args(name),
+                                display=display,
+                                user=self.get_current_user()))
+
+def display(stat):
```

Comment:
Is this used on `ont_flowcell.html`?

Reply:
9dda18f51d82d37e5c6c733e160ac4dc004ff797